api/lib/infrastructure/repositories/campaign-report-repository.js

```diff
@@ -56,8 +56,11 @@ module.exports = {
         'users.id AS "creatorId"',
         'users.firstName AS creatorFirstName',
         'users.lastName AS creatorLastName',
+        knex.raw('COUNT(*) FILTER (WHERE "campaign-participations"."id" IS NOT NULL) OVER (partition by "campaigns"."id") AS "participationsCount"'),
+        knex.raw('COUNT(*) FILTER (WHERE "campaign-participations"."id" IS NOT NULL AND "campaign-participations"."isShared" IS TRUE) OVER (partition by "campaigns"."id") AS "sharedParticipationsCount"'),
```

Comment:
**question (non-blocking) :** C'est auto magique de récupérer le bon `count` pour la bonne campagne ?

Reply:
je pense que c'est grâce au `OVER campaign.id` .

Reply:
C'est précisément le rôle du `partition by "campaigns"."id"`.
La fonction fenêtrée en PostgreSQL est une feature qui permet de faire des calculs agrégés en regroupant par paquet de son choix.
L'écriture en gros c'est :
fonction_aggregation() OVER (partition by critere_de_regroupement)

Donc je dirais pas que c'est auto-magique puisque c'est le rôle de la feature 😅

Reply:
Du coup le partition by se débrouille pour récupérer le bon ensemble de participations à partir de la campagnId de la ligne ?

Reply:
Par exemple naturellement j'aurais vu un truc `(WHERE "campaign-participations"."id" IS NOT NULL AND campaignId = campaigns.id)`, je pense que comprend juste pas bien comment marche le partition by avec le over.

Quand je lis ça  OVER (partition by "campaigns"."id") => je comprend "fait des ensembles en te basant sur le campaignId" mais je ne vois pas comment on lui dit quel ensemble utilisé